symfit/contrib/interactive_guess/interactive_guess.py

```diff
@@ -206,4 +239,126 @@ def __str__(self):
+        kde = gaussian_kde(projected_data)
+
+        xx, yy = np.meshgrid(self.ig._x_points[x.name], self.ig._y_points[y.name])
+        x_grid = xx.flatten()
+        y_grid = yy.flatten()
+
+        contour_grid = kde.pdf(np.column_stack((x_grid, y_grid)).T)
+        # This is an fugly kludge, but it seems nescessary to make low density
+        # areas show up.
+        if self.ig.log_contour:
+            contour_grid = np.log(contour_grid)
+            vmin = -7
+        else:
+            vmin = None
+        ax.contourf(xx, yy, contour_grid.reshape(xx.shape),
+                    50, vmin=vmin, cmap='Blues')
+
+    def plot_model(self, proj, ax):
+        """
+        Plots the model proposed for the projection proj on ax.
+        """
+        x, y = proj
+        evaluated_model = self.ig._eval_model()
+        y_vals = getattr(evaluated_model, y.name)
+        x_vals = self.ig._x_grid[x.name]
+        plot = ax.errorbar(x_vals, y_vals, xerr=0, yerr=0, c='red')
+        return plot
+
+    def update_plot(self, indep_var, dep_var):
+        """
+        Updates the plot of the proposed model.
+        """
+        evaluated_model = self.ig._eval_model()
+        y_vals = getattr(evaluated_model, dep_var.name)
+        x_vals = self.ig._x_grid[indep_var.name]
+
+        x_plot_data = []
+        y_plot_data = []
+        y_plot_error = []
+        # TODO: Numpy magic
+        # We need the error interval for every plotted point, so find all
+        # the points plotted at x=x_i, and do some statistics on those.
+        # Since all the points are on a grid made by meshgrid, the error
+        # in x will alwys be 0.
+        for x_val in self.ig._x_points[indep_var.name]:
+            # We get away with this instead of digitize because x_vals is
+            # on a grid made with meshgrid
+            idx_mask = x_vals == x_val
+            xs = x_vals[idx_mask]
+            x_plot_data.append(xs[0])
+            ys = y_vals[idx_mask]
+            y_plot_data.append(np.mean(ys))
+            y_error = np.percentile(ys, self.ig.percentile)
+            y_plot_error.append(y_error)
+
+        x_plot_data = np.array(x_plot_data)
+        y_plot_data = np.array(y_plot_data)
+        y_plot_error = np.array(y_plot_error)
+
+        xs = np.column_stack((x_plot_data, x_plot_data))
+        yerr = y_plot_error + y_plot_data[:, np.newaxis]
+        y_segments = np.dstack((xs, yerr))
+        plot_line, caps, error_lines = self.ig._plots[(indep_var, dep_var)]
+        plot_line.set_data(x_plot_data, y_plot_data)
+        error_lines[1].set_segments(y_segments)
+
+
+class InteractiveGuess2D(InteractiveGuess):
+    def __init__(self, *args, **kwargs):
+        import warnings
+        # Deprecated as of 01/06/2017
+        warnings.warn("InteractiveGuess2D is deprecated in favor of InteractiveGuess")
+        # raise DeprecationWarning("InteractiveGuess2D is deprecated in favor of InteractiveGuess")
```

Comment:
Why not this DeprecationWarning so it is a proper subclass of DeprecationWarning rather than just a warning?

Reply:
Consider using the `@deprecated` decorator from this from `support.py`. It makes this process very easy.

Reply:
All these comments should still be removed.